winch/codegen/src/codegen/mod.rs

```diff
@@ -1417,6 +1417,30 @@ where
         }
         Ok(())
     }
+
+    pub fn emit_load_lane(&mut self, arg: &MemArg, lane: u8, size: OperandSize) -> Result<()> {
+        let dst = self.context.pop_to_reg(self.masm, None)?;
+        if let Some(addr) = self.emit_compute_heap_address(&arg, size)? {
+            let src = self.masm.address_at_reg(addr, 0)?;
+            self.masm.load_lane(writable!(dst.reg), src, lane, size)?;
+            self.context.stack.push(dst.into());
+            self.context.free_reg(addr);
+        }
+
+        Ok(())
+    }
+
+    pub fn emit_store_lane(&mut self, arg: &MemArg, lane: u8, size: OperandSize) -> Result<()> {
+        let src = self.context.pop_to_reg(self.masm, None)?;
+        if let Some(addr) = self.emit_compute_heap_address(&arg, size)? {
+            let dst = self.masm.address_at_reg(addr, 0)?;
+            self.masm.store_lane(src.reg, dst, lane, size)?;
+            self.context.free_reg(addr);
+            self.context.free_reg(src);
+        }
+
+        Ok(())
+    }
```

Comment:
Instead of introducing new load/stored methods, I'd suggest calling `wasm_load` and `wasm_store`. This approach would probably require modifying the existing `LoadKind` definition in case the current definition can't represent the semantics of lane loading/storing. That's the approach we've followed for other vector loads (e.g., `load_splat`)

1 main reason:
- Wasm loads/stores are critical, from the sandboxing perspective, and we're trying to keep the implementation as tight as possible, reusing the existing methods makes auditing them and maintaining them much easier.